cfme/tests/test_db_migrate.py

```diff
@@ -70,38 +71,79 @@ def appliance_preupdate(temp_appliance_preconfig_funcscope_upgrade, appliance):
     return temp_appliance_preconfig_funcscope_upgrade
 
 
+def guess_the_db_format(basename):
+    if basename.endswith('dumpall'):
+        return 'pg_dumpall'
+    elif basename.endswith('backup') or basename.endswith('dump'):
+        return 'pg_dump'
+    else:
+        raise Exception("Couldn't guess the db format")
+
+
 def download_and_migrate_db(app, db_url):
+    def fetch(src, dst):
+        result = app.ssh_client.run_command(
+            f'curl  --fail -S -o "{dst}" "{src}"', timeout=15)
+        assert result.success, f"Failed to download {src}:\n{result.output}"
+
     # Download the database
     logger.info("Downloading database: {}".format(db_url))
     url_basename = os_path.basename(db_url)
     loc = "/tmp/"
-    result = app.ssh_client.run_command(
-        'curl -o "{}{}" "{}"'.format(loc, url_basename, db_url), timeout=30)
-    assert result.success, "Failed to download database: {}".format(result.output)
-    # The v2_key is potentially here
-    v2key_url = os_path.join(os_path.dirname(db_url), "v2_key")
+    v2key_url = os_path.join(os_path.dirname(db_url), "v2_key.bak")
+    database_yml_url = os_path.join(os_path.dirname(db_url), "database.yml")
+    db_format = guess_the_db_format(url_basename)
+
+    fetch(db_url, f'{loc}{url_basename}')
+
     # Stop EVM service and drop vmdb_production DB
     app.evmserverd.stop()
+    # Invalidate some of the cache
+    # as it was be causing problems
+    # when running more than one test with
+    # the same vm.
+    app.__dict__.pop('rest_api', None)
```

Comment:
Q: Will this be done only on a temp appliance?

Reply:
I don't understand the question. It will be done on the app that is provided to the `download_and_migrate`. Any appliance that is given to this method will get the db wiped and replaced with some completely different one. I don't see for what reason one would like to keep the cached api results.

Reply:
Ah my original concern was that this would take away the `rest_api` endpoint for any appliance passed to the method. But I see that that's not the case. Thanks :+1: